src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -1399,24 +1401,26 @@ protected void ExecuteIndexing()
                         while (true)
                         {
 
-                            WaitHandle.WaitAny(new [] {_mre.WaitHandle, _rollingEvent.WaitHandle, _indexingProcessCancellationTokenSource.Token.WaitHandle});
+                            var result = WaitHandle.WaitAny(new [] {_mre.WaitHandle, _rollingEvent.WaitHandle, _indexingProcessCancellationTokenSource.Token.WaitHandle});
                             _indexingProcessCancellationTokenSource.Token.ThrowIfCancellationRequested();
                             
                             if (_indexDisabled)
                                 return;
-
-                            _rollingEvent.Reset();
```

Comment:
So now there is no `_rollingEvent.Reset()` call, is that fine?

Reply:
We did `_rollingEvent.Set()` later on.. So I simply removed them both and if the event was called `_rollingEvent` it still will be set and the index replace will be retried.

Reply:
Okay. Just to make sure and make you will think about it for a moment - will it still work correctly if the index definition will be changed again with rolling deployment enabled?

Reply:
Not sure that I follow exactly what you mean here, but I've added a test to verify the behavior when the index is changed multiple times before the rolling is over

Reply:
That's exactly what I meant. Perfect.